src/main.rs

```diff
@@ -786,23 +840,24 @@ fn main() -> Result<()> {
                 euclidean_distance
             };
 
-            let sort = match sub_m.is_present("seed") {
+            let mut sort = match sub_m.is_present("seed") {
                 false => closest_to_songs,
                 true => song_to_song,
             };
-            if sub_m.is_present("dedup") {
-                library.queue_from_current_song_custom(
+            let no_dedup = sub_m.is_present("no-dedup");
+            if sub_m.is_present("entire") {
+                library.queue_from_current_playlist(
                     number_songs,
-                    &distance_metric,
-                    sort,
-                    true,
+                    forest_distance,
```

Comment:
Not sure how many more TODOs you have in mind for this PR, but FWIW I would really like this to respect the `--distance` argument! When I was testing the original PR I found that the quality of the playlists that the isolation forest generated were not as good as the euclidean distance, and I would rather keep using that.
(I've been using that PR since then with `--distance euclidean` and have really been enjoying being able to pick a few different songs and see what it comes up with from there.)

Reply:
indeed, good idea, I'll restructure this a bit.

Just a heads-up, the original PR used `ForestOptions::default()`, and I used the ones recommended in the article (I think?), so maybe it will give better results?

Reply:
I just completed the PR, how does it look now? :)

Reply:
okay, now I fixed it for good - I don't know what happened 🤦
It should look good now. I'll refactor it a bit to accommodate the changes I made for blissify 0.4.0, but functionality-wise, it shouldn't move.